compiler/rustc_const_eval/src/transform/check_consts/qualifs.rs

```diff
@@ -172,7 +173,8 @@ impl Qualif for NeedsNonConstDrop {
 
         if !matches!(
             impl_src,
-            ImplSource::Builtin(_) | ImplSource::Param(_, ty::BoundConstness::ConstIfConst)
+            ImplSource::Builtin(BuiltinImplSource::Misc, _)
```

Comment:
all builtin destruct impls are not const :sweat_smile: i would like us to still use all `ImplSource::Builtin` in this match, or even make this exhaustive xx

Reply:
I'm gonna leave this here for now

Reply:
> all builtin destruct impls are not const

Also, what do you mean? They're definitely const in the *old* solver?

Reply:
this is a question of what exactly this fn does... missed the double negation of 1) returning `true` if it is not const and 2) the `!` infront of the match.

This automatically considers all builtin impls const if their source is `Misc` which feels like a footgun to me :thinking: anyways, const trait is a bigger issue ^^

Reply:
Oh, sorry, I didn't understand what you meant there. Yeah, shrug... not really changing behavior here